src/test/java/com/yelp/nrtsearch/server/backup/BackupHelper.java

```diff
@@ -0,0 +1,300 @@
+    return fileCompressAndUploaderWithNoTar;
+  }
+
+  public VersionManager getVersionManager() {
+    return versionManager;
+  }
+
+  public AmazonS3 getS3() {
+    return s3;
+  }
+
+  public Path createIndexDir(Map<String, String> fileAndContents, Archiver archiver)
+      throws IOException {
+    Path indexDir = IndexArchiver.getIndexDataDir(archiverDirectory);
+    Path stateDir = IndexArchiver.getIndexStateDir(archiverDirectory);
+    if (!Files.exists(indexDir)) {
+      Files.createDirectories(indexDir);
+    }
+    if (!Files.exists(stateDir)) {
+      Files.createDirectories(stateDir);
+    }
+    for (Map.Entry<String, String> entry : fileAndContents.entrySet()) {
+      for (Path path : List.of(indexDir, stateDir)) {
+        Path file = Paths.get(path.toString(), entry.getKey());
+        if (!Files.exists(file)) {
+          Files.writeString(Files.createFile(file), entry.getValue(), StandardCharsets.UTF_8);
+        }
+      }
+    }
+    if (archiver != null && archiver instanceof IndexArchiver) {
+      return archiverDirectory;
+    } else {
+      return indexDir;
+    }
+  }
+
+  public String uploadBlessAndValidate(
+      Map<String, String> filesAndContents, Archiver archiver, String service, String resource)
+      throws IOException {
+    Path indexDir = createIndexDir(filesAndContents, archiver);
+    String versionHash =
+        archiver.upload(
+            service, resource, indexDir, filesAndContents.keySet(), Collections.emptyList(), true);
+    assertEquals(true, archiver.blessVersion(service, resource, versionHash));
+    // files are uploaded
+    for (String file : filesAndContents.keySet()) {
+      assertEquals(
+          true,
+          getS3().doesObjectExist(bucketName, getS3ResourceKey(service, resource, file, archiver)));
+    }
+    // versionHash is uploaded
+    assertEquals(
+        true,
+        getS3()
+            .doesObjectExist(
+                bucketName, getS3ResourceKey(service, resource, versionHash, archiver)));
+
+    // Extra check for IndexArchiver only - check state files uploaded under versionHash
+    if (archiver != null && archiver instanceof IndexArchiver) {
+      assertEquals(
+          true,
+          getS3()
+              .doesObjectExist(
+                  bucketName,
+                  String.format(
+                      "%s/%s/%s",
+                      service, IndexArchiver.getIndexStateResourceName(resource), versionHash)));
+      resource = IndexArchiver.getIndexDataResourceName(resource);
+    }
+    // latest versionHash exists
+    assertEquals(
+        true,
+        getS3()
+            .doesObjectExist(
+                bucketName, String.format("%s/_version/%s/_latest_version", service, resource)));
+    return versionHash;
+  }
+
+  private String getS3ResourceKey(String service, String resource, String file, Archiver archiver) {
+    if (archiver != null && archiver instanceof IndexArchiver) {
```

Comment:
This is a test method but still feels like a weird way to check if the resource is index data or not. It might be clearer to pass a more relevant parameter than Archiver here

Reply:
Okay, I changed the signature of this private test method to take in a boolean flag to indicate if it is an IndexArchiver